miio/integrations/vacuum/dreame/dreamevacuum_miot.py

```diff
@@ -230,54 +409,179 @@ def status(self) -> DreameVacuumStatus:
 
     @command()
     def identify(self) -> None:
         """Locate the device (i am here)."""
-        return self.send_action(17, 1)
+        return self.call_action("locate")
 
     @command()
     def reset_mainbrush_life(self) -> None:
         """Reset main brush life."""
-        return self.send_action(26, 1)
+        return self.call_action("reset_mainbrush_life")
 
     @command()
     def reset_filter_life(self) -> None:
         """Reset filter life."""
-        return self.send_action(27, 1)
+        return self.call_action("reset_filter_life")
 
     @command()
     def reset_sidebrush_life(self) -> None:
         """Reset side brush life."""
-        return self.send_action(28, 1)
+        return self.call_action("reset_sidebrush_life")
+
+    @command()
+    def play_sound(self) -> None:
+        """Play sound."""
+        return self.call_action("play_sound")
+
+    @command()
+    def fan_speed(self):
+        """Return fan speed."""
+        dreame_vacuum_status = self.status()
+        fanspeed = dreame_vacuum_status.cleaning_mode
+        if not fanspeed or fanspeed.value == -1:
+            _LOGGER.warning("Unknown fanspeed value received")
+            return
+        return {fanspeed.name: fanspeed.value}
+
+    @command(click.argument("speed", type=int))
+    def set_fan_speed(self, speed: int):
+        """Set fan speed.
+
+        :param int speed: Fan speed to set
+        """
+        fanspeeds_enum = self.get_fanspeeds()
+        fanspeed = None
+        if not fanspeeds_enum:
+            return
+        try:
+            fanspeed = fanspeeds_enum(speed)
+        except ValueError:
+            return None
+        click.echo(f"Setting fanspeed to {fanspeed.name}")
+        return self.set_property("cleaning_mode", fanspeed.value)
+
+    @command()
+    def fan_speed_presets(self) -> Dict[str, int]:
+        """Return dictionary containing supported fan speeds."""
+        fanspeeds_enum = self.get_fanspeeds()
+        if not fanspeeds_enum:
+            return {}
+        return _enum_as_dict(fanspeeds_enum)
+
+    @command()
+    def waterflow(self):
+        """Get water flow setting."""
+        dreame_vacuum_status = self.status()
+        waterflow = dreame_vacuum_status.water_flow
+        if not waterflow or waterflow.value == -1:
+            _LOGGER.warning("Unknown waterflow value received")
+            return
+        return {waterflow.name: waterflow.value}
+
+    @command(click.argument("value", type=int))
+    def set_waterflow(self, value: int):
+        """Set water flow.
+
+        :param int value: Water flow value to set
+        """
+        mapping = self._get_mapping()
+        if "water_flow" not in mapping:
+            return None
+        waterflow = None
+        try:
+            waterflow = WaterFlow(value)
+        except ValueError:
+            return None
+        click.echo(f"Setting waterflow to {waterflow.name}")
+        return self.set_property("water_flow", waterflow.value)
```

Comment:
Do you think these are necessary when the information is already available when you always call `status()` and access it from there?

Reply:
`de waterflow(self)`It is aligned with roborock implementation

Reply:
Okay, let's keep it then for now :+1: